### ISSUE TITLE
Add autodeployment for the Doxygen web-page

### ISSUE THREAD

**Issue by mkstoyanov** [ORNL/TASMANIAN#417]
* currently the web-page in `gh-pages` is updated manually
* add an automated procedure to rebuild the pages upon merging a pull request

**Comment by dalg24** [comment]
Partially addressed by #425 and #432

**Comment by dalg24** [comment]
Remaining issue if the documentation generated is the same as the one that is already online:
```
On branch gh-pages
Untracked files:
	./

nothing added to commit but untracked files present
```
Full log [here](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/TASMANIAN/detail/master/8/pipeline/)

**Comment by mkstoyanov** [comment]
We should also squash the commits, one the process is polished, no need to keep 2 - 3MB per PR.

**Comment by mkstoyanov** [comment]
With #438 and #439 the process is officially working. Thanks @dalg24

**Comment by mkstoyanov** [comment]
Minor adjustments left:
* the web-page build on merge, but the regular tests run again
    * do not repeat the CI when the PR is merged
* the gh-pages branch will keep growing, especially due to binary files such as formulas and charts
    * squash the last two commits and force-push to the gh-pages

**Comment by mkstoyanov** [comment]
squashing commits is addressed in #441

**Comment by mkstoyanov** [comment]
It works and it's alive! https://ornl.github.io/TASMANIAN/

Thanks, @dalg24 !
